esx_service/utils/vmdk_utils.py

```diff
@@ -100,8 +100,11 @@ def get_datastores():
     'url-name' is the last element of datastore URL (e.g. 'vsan:572904f8c031435f-3513e0db551fcc82')
     'dockvol-path; is a full path to 'dockvols' folder on datastore
     """
-    if datastores == None:
-        init_datastoreCache()
+    # if datastores == None:
+    #     init_datastoreCache()
+    # we should not use cached value, user may change datastore name on the fly
```

Comment:
the problem is that init_datastoreCache is a an  expensive operation and it is called way too many times, often many times in the same call chain. The correct way to fix it would have been passing datastore list, but it was a very invasive change so the cache was introduced.  Let's discuss how to test and what are the other ways to fix the issue, rather than refetch all DS list every time